### ISSUE TITLE
Add shuck as alternative to shellcheck as code-block-tools built-in-tool

### ISSUE THREAD

**Issue by hofbi** [rvben/rumdl#655]
### Problem or motivation

https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md#built-in-tools is only supporting shellcheck. But there is also https://github.com/ewhauser/shuck as a Rust based replacement for shellcheck.

### Proposed solution

Add https://github.com/ewhauser/shuck as supported built-in tool.

### Feature category

Improve existing rule

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Good one @hofbi, will have a quick look if more tools are missing.

**Comment by rvben** [comment]
Quick status update, @hofbi 👍

I dug into wiring shuck up as a built-in and hit a blocker on shuck's side. rumdl's code-block tools feed each block to the tool over **stdin** (that's how `shellcheck`, `shfmt`, etc. work as built-ins). But shuck's `check` subcommand doesn't read stdin — `shuck check -` treats `-` as a filename rather than reading the piped script, and there's no `--stdin` flag. Stdin only works in shuck's shellcheck-compat mode (via an env var) or in `shuck format`, neither of which a built-in tool definition can express. So I can't ship a `shuck` built-in that actually runs today.

I've filed an upstream request for native `shuck check -` stdin support: [ewhauser/shuck#1123](https://github.com/ewhauser/shuck/issues/1123). The moment shuck can read a script from stdin, the rumdl side is trivial — it's a one-line registry entry (`shuck check --output-format concise -`) plus docs, and I'll add it.

Keeping this open as the tracking issue. Thanks again for the suggestion — and if you want to nudge the stdin support upstream, a 👍 on #1123 wouldn't hurt.

**Comment by hofbi** [comment]
Thanks for driving this. Now that you mentioned shuck check and shuck fmt, do you think it makes sense to add both of them? One as formatter and alternative to shfmt and the other as linter as alternative to Shellcheck?

**Comment by rvben** [comment]
Yeah, sure. That'd be easy to add when stdin is properly supported. 👍

**Comment by soy-chrislo** [comment]
Good timing on this one: the stdin blocker just cleared. `ewhauser/shuck#1123` got merged and closed a few hours ago, and shuck v0.0.43 (released today too) finally has `shuck check -` reading properly from stdin, plus a `--stdin-filename` flag.

I grabbed v0.0.43 and ran it against a few shell snippets. `shuck check --output-format concise -` prints to stdout, exits 1 when it finds something and 0 on a clean script, and the output lines up with the diagnostic format rumdl already parses for the other built-in tools, so nothing extra needed there.

I'd like to pick this up, both `check` and `format` as discussed here back in June, rather than shipping just the lint half and circling back later. Before starting, a couple of scope questions for @rvben:

- `check` is a one-line registry entry, same pattern as `shellcheck`.
- `format` is a bit different: it's flagged experimental upstream right now and needs `SHUCK_EXPERIMENTAL=1` to run, which none of the current built-in tools need since they all just run directly. Wire it in as is, or hold off until shuck marks it stable?

Happy to put the PR together for both, if this can be handed off.

**Comment by rvben** [comment]
Thanks for verifying shuck v0.0.43 and for the offer, handing it off gladly. Let's split the work: please go ahead with the `shuck` (lint) entry now, `shuck check --output-format concise -` matches what the output parser expects, plus the paired docs entry (`make sync-code-block-tools`) and an execution-harness test per the repo conventions. For `format`, let's hold off until upstream drops the `SHUCK_EXPERIMENTAL` gate rather than adding env-var plumbing to built-in tool definitions for an experimental feature; this issue stays open to track that half.

**Comment by rvben** [comment]
`shuck:format` shipped in **v0.2.36**, completing the second half of this request. The lint half (`shuck`) shipped earlier via #729.

What's available now:
- `shuck` → `shuck check --output-format concise -` (lint)
- `shuck:format` → `shuck format -` (format)
- a bare `shuck` in a format slot resolves to `shuck:format` automatically

The `SHUCK_EXPERIMENTAL` gate that had blocked `shuck format` is gone as of shuck 0.0.45, so no env-var plumbing was needed. Requires shuck >= 0.0.43 for stdin support.

Thanks @hofbi for the request and @soy-chrislo for the lint-half PR. Both `check` and `format` are supported now, so closing this out.
